lib/selector/parser/ast.go

```diff
@@ -214,6 +281,10 @@ func (node NotNode) Evaluate(labels Labels) bool {
 	return !node.Operand.Evaluate(labels)
 }
 
+func (node *NotNode) AcceptVisitor(v Visitor) {
+	node.Operand.AcceptVisitor(v)
```

Comment:
It doesn't matter for your use case, but I think, in general, you should do `v.Visit(node)` before visiting the operand.

(Same for the other non-leaf nodes)

Reply:
d'oh yes, I meant to do this but forgot.